app/models/users_groups_role.py

```diff
@@ -1,23 +1,43 @@
+import random
+
+from citext import CIText
+from flask_jwt_extended import current_user
+
+from app.api.helpers.errors import ForbiddenError
+from app.api.helpers.mail import send_email_group_role_invite
 from app.models import db
+from app.models.group import Group
+from app.models.role import Role
+from app.settings import get_settings
+
+
+def generate_hash():
+    hash_ = random.getrandbits(128)
+    return str(hash_)
 
 
 class UsersGroupsRoles(db.Model):
     __tablename__ = 'users_groups_roles'
     __table_args__ = (
         db.UniqueConstraint(
-            'user_id', 'group_id', 'role_id', name='uq_uer_user_group_role'
+            'email', 'user_id', 'group_id', 'role_id', name='uq_uer_user_group_role'
```

Comment:
Wrong. You need to add another unique constraint with email instead of user_id

There needs to be 2 unique constraints